oracle/oracle.go

```diff
@@ -441,3 +453,152 @@ func (o *oracleOps) Delete(volumeID string) error {
 	}
 	return nil
 }
+
+// Attach volumeID, accepts attachOptions as opaque data
+// Return attach path.
+func (o *oracleOps) Attach(volumeID string, options map[string]string) (string, error) {
+	o.mutex.Lock()
+	defer o.mutex.Unlock()
+
+	devices, err := o.FreeDevices([]interface{}{}, "")
+	if err != nil {
+		return "", err
+	}
+
+	for _, device := range devices {
+		attachVolReq := core.AttachVolumeRequest{
+			AttachVolumeDetails: core.AttachParavirtualizedVolumeDetails{
+				InstanceId:  common.String(o.instance),
+				VolumeId:    common.String(volumeID),
+				Device:      common.String(device),
+				IsShareable: common.Bool(false),
+				IsReadOnly:  common.Bool(false),
+			},
+		}
+
+		attachVolResp, err := o.compute.AttachVolume(context.Background(), attachVolReq)
+		if err != nil {
+			if strings.Contains(err.Error(), "is already in use") {
```

Comment:
Yes, when already used devicePath is used again, below error msg is returned by cloud provider

> Error returned by Compute Service. Http Status Code: 409. Error Code: Conflict. Opc request id: abc108c45d2d8bca330e7f0653150ba2/A45C10BA4BF42BE95E0901102DE9C190/0154264236916716F99C9557D8946775. Message: The volume cannot be attached to the instance ocid1.instance.oc1.us-sanjose-1.anzwuljrm7v3lmackrb4m37ekyndtty4vwagoawbanxtrxlavhac2krkexwq because the specified device attribute /dev/oracleoci/oraclevdb is already in use. Specify a different device attribute and try again.